OpenRA.Game/Server/Server.cs

```diff
@@ -791,9 +791,12 @@ void OutOfSync(int frame)
 
 			// Make sure the written file is not valid
 			// TODO: storing a serverside replay on desync would be extremely useful
-			recorder.Metadata = null;
+			if (recorder != null)
```

Comment:
Why is this needed?

Reply:
without metadata the server isn't aware of the state of players

Reply:
Though I guess here specifically, since we now also handle sync orders without the existence of recorder we now need to deal with it being null in more places

Reply:
Why/where does this PR change the cases where `recorder` is null?

Reply:
You can see the changelog. It's rather short. 

For the server to know what the state of the game is, it needs to handle sync orders unconditionaly. That is regardless of the existance of recorder. But if we handle unconditionaly, then all the extra code we run needs to be aware of the posibility that recorder is null

Reply:
Ah right, we call `HandleSyncOrder` now.